### DISCUSSION TITLE
Cannot run compiled executable after downloading from GitHub releases

### DISCUSSION THREAD

**Discussion by jsjoeio** [denoland/deno#10382] [Q&A] [answered]
I wrote a CLI in Deno ([source](https://github.com/jsjoeio/jp-courses-install)). 

## It works when...

I compile locally and run the executable.

![image](https://user-images.githubusercontent.com/3806031/116101074-62d40500-a662-11eb-9ade-f882cba9275c.png)

## It doesn't work when...

I download the executable as an asset from [GitHub releases](https://github.com/jsjoeio/jp-courses-install/releases/tag/0.0.1) and run locally.

![image](https://user-images.githubusercontent.com/3806031/116101428-a890cd80-a662-11eb-9255-24933eb6ec8d.png)

Do I need to sign the executables or do something else to get them to work out-of-the-box?

P.S. - would be more than happy to contribute to the ["Compiling Executables"](https://deno.land/manual@main/tools/compiler) docs once I figure this out.

**Comment by laughedelic** [comment, marked as answer]
Maybe you just need to make it an executable? `chmod +x ./filename`

**Reply by jsjoeio** [reply]
Wow 🤦‍♂️ That was it. Guess I'll have to do that before uploading the assets to the GitHub release.

Though, I did have to grant specific approval. I was shown this message:
![image](https://user-images.githubusercontent.com/3806031/116102433-906d7e00-a663-11eb-901d-1e62a952214a.png)

Then I had to go into System Preferences > Security > General and click "Allow Anyway" and then click Open after running again.

**Reply by jsjoeio** [reply]
Unfortunately, this did not do the trick :( 

I made the files executable with `chmod +x` before uploading to GitHub releases and even after downloading again, I still get permission errors.

**Reply by laughedelic** [reply]
I don't think that after uploading an executable to GitHub releases will preserve its executable status. It's just a file.

**Reply by jsjoeio** [reply]
Ah okay. I guess I thought it would since if you make a file executable and commit to a repo, it's executable later.

How am I supposed to upload it in a way that preserves its executable status?

**Reply by jsjoeio** [reply]
I chatted with a couple folks on the Deno Discord. They suggested try packaging it as a zip or tarball, which may preserve file permissions

**Reply by jsjoeio** [reply]
That ended up working! I zipped it up and now when you unzip, it has the right permissions (at least when I tried on Linux).

https://github.com/jsjoeio/jp-courses-install/releases/tag/0.0.3

**Reply by TotemPear** [reply]
How would you make it so that you don't need to zip it then? Because it's a bit annoying to have to do that :/
